### PR TITLE
Fixes #32422 - cannot create repo with http proxy with auth creds

### PR DESCRIPTION
**Pull request by ianballou**
Moves the HTTP proxy username and password out of the URL and into the RpmRpmRemote parameters.

To test:

1) Ensure you're on Pulpcore 3.11
2) Create an HTTP proxy that has a username and password
3) Create a repository with that HTTP proxy attached and see the error in the redmine

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

**Comment by ianballou** [comment]
I realized that `remote_needs_updates?` is failing because Pulp no longer returns  the `proxy_username` or `proxy_password` for security reasons. We'll need another way to tell if those have changed.

**Comment by ianballou** [comment]
Looks like a couple tests need updating. Clearing out the tests that need re-recorded VCRs first.

**Comment by jturel** [comment]
I can't reproduce the original issue:

```
hammer http-proxy create --name TestProxy --username admin --password changeme --url http://localhost:8181
hammer product create --name TestProduct --organization-id=1
hammer repository create --name TestRepoBeforeChange --organization-id=1 --http-proxy-id=1 --product=TestProduct --content-type=yum
```

Do my steps look OK?

```
[root@centos7-katello-devel katello]# rpm -qa | grep pulp | sort
pulp-client-1.0-1.noarch
pulpcore-selinux-1.2.4-1.el7.x86_64
python3-pulp-ansible-0.7.1-1.el7.noarch
python3-pulp-certguard-1.2.0-1.el7.noarch
python3-pulp-container-2.5.2-2.el7.noarch
python3-pulpcore-3.11.0-2.el7.noarch
python3-pulp-deb-2.11.1-1.el7.noarch
python3-pulp-file-1.6.0-1.el7.noarch
python3-pulp-rpm-3.10.0-1.el7.noarch
tfm-rubygem-smart_proxy_pulp-3.0.0-1.fm2_5.el7.noarch

```

**Comment by jturel** [comment]
Sorry for the accidental close. FYI there was a rubocop error: https://ci.theforeman.org/blue/organizations/jenkins/katello-pr-test/detail/katello-pr-test/7064/pipeline

**Comment by ianballou** [comment]
@jturel I see the issue: `:proxy_url=>nil`

It looks like `root.http_proxy&.full_url` is nil. I didn't test this through hammer, but it looks like the http proxy isn't being associated.

File: app/models/katello/concerns/setting_extensions.rb [hunk 1]
```diff
@@ -17,7 +17,12 @@ def recalculate_errata_status
 
       def update_global_proxies
         if saved_change_to_attribute?(:value) && name == 'content_default_http_proxy'
-          repos = RootRepository.with_global_proxy.collect(&:library_instance).uniq
+          repos = []
+          RootRepository.with_global_proxy.each do |root_repo|
```
Comment (jturel) [inline app/models/katello/concerns/setting_extensions.rb:21]: Is this AR query equivalent to this logic you implemented here?

`Katello::Repository.joins(:root).merge(Katello::RootRepository.with_global_proxy).where.not(remote_href: nil).where(library_instance_id: nil)`

```
SELECT "katello_repositories".* FROM "katello_repositories" INNER JOIN "katello_root_repositories" ON 
"katello_root_repositories"."id" = "katello_repositories"."root_id" WHERE 
"katello_root_repositories"."http_proxy_policy" = 'global_default_http_proxy' AND 
"katello_repositories"."remote_href" IS NOT NULL AND "katello_repositories"."library_instance_id" IS NULL
```

Comment (ianballou) [reply app/models/katello/concerns/setting_extensions.rb:21]: @jturel looks like it accomplishes the same thing, yeah

Comment (ianballou) [reply app/models/katello/concerns/setting_extensions.rb:21]: Think I should substitute it in for this and the other similar spot?

Comment (ianballou) [reply app/models/katello/concerns/setting_extensions.rb:21]: I realize that in the other case, `repositories_with_proxy` returns an array of repos. That one might need to be a bit different.

**Review by jturel — APPROVED** [review]
ACK

Nice!

**Pull request merged**